src/main/java/com/github/codeidoscope/ServerConnection.java

```diff
@@ -0,0 +1,58 @@
+package com.github.codeidoscope;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class ServerConnection implements ServerConnectionWrapper {
+    private ServerSocket serverSocket;
+    private Socket socket;
+    InputStream inputStream;
+    OutputStream outputStream;
+    private int portNumber;
+
+    @Override
+    public void openConnection(int portNumber) {
+        this.portNumber = portNumber;
+        try {
+            serverSocket = new ServerSocket(portNumber);
+            socket = serverSocket.accept();
+            inputStream = socket.getInputStream();
+            outputStream = socket.getOutputStream();
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Override
+    public String getInput() {
+        try {
+            return new BufferedReader(new InputStreamReader(inputStream)).readLine();
+        } catch (IOException e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Override
+    public void sendOutput(String message) {
+        new PrintWriter(outputStream, true).println(message);
+    }
+
+    @Override
+    public void closeConnection() {
+        System.out.println("Closing connection - Bye! \uD83D\uDC4B");
+        try {
+            inputStream.close();
+            outputStream.close();
+            socket.close();
+            serverSocket.close();
+        } catch (IOException e) {
+            throw new RuntimeException();
```

Comment:
As you are doing nothing with the exception at this level, you could use a checked exception